packages/emitter-framework/src/python/components/function-declaration/function-declaration.test.tsx

```diff
@@ -1,23 +1,351 @@
+        pass
+      
+      `);
+  });
+
+  it("creates a function with a custom name", async () => {
+    const { program, getName } = await Tester.compile(t.code`
+      op ${t.op("getName")}(id: string): string;
+    `);
+
+    expect(getOutput(program, [<FunctionDeclaration name="new_name" type={getName} />]))
+      .toRenderTo(`
+      def new_name(id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates a function appending extra parameters with raw params provided", async () => {
+    const { program, createPerson } = await Tester.compile(t.code`
+      op ${t.op("createPerson")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <FunctionDeclaration
+          type={createPerson}
+          parameters={[
+            { name: "name", type: "str" },
+            { name: "age", type: "float" },
+          ]}
+        />,
+      ]),
+    ).toRenderTo(`
+      def create_person(name: str, age: float, id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates a function prepending extra parameters with raw params provided", async () => {
+    const { program, createPerson } = await Tester.compile(t.code`
+      op ${t.op("createPerson")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <FunctionDeclaration
+          type={createPerson}
+          parameters={["name", "age"]}
+          parametersMode="prepend"
+        />,
+      ]),
+    ).toRenderTo(`
+      def create_person(name, age, id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates a function replacing parameters with raw params provided", async () => {
+    const { program, createPerson } = await Tester.compile(t.code`
+      op ${t.op("createPerson")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <FunctionDeclaration
+          type={createPerson}
+          parameters={["name", "age"]}
+          parametersMode="replace"
+        />,
+      ]),
+    ).toRenderTo(`
+      def create_person(name, age) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates a function with defaults in raw parameter descriptors", async () => {
+    const { program, createPerson } = await Tester.compile(t.code`
+      op ${t.op("createPerson")}(id: string): string;
+    `);
+
+    expect(
+      getOutput(program, [
+        <FunctionDeclaration
+          type={createPerson}
+          parameters={[
+            { name: "limit", type: "int", default: <py.Atom jsValue={10} /> },
+            { name: "verbose", type: "bool", default: <py.Atom jsValue={true} /> },
+          ]}
+          parametersMode="prepend"
+        />,
+      ]),
+    ).toRenderTo(`
+      def create_person(limit: int = 10, verbose: bool = True, id: str) -> str:
+        pass
+      
+      `);
+  });
+
+  it("creates a function building parameters from a model via parametersModel (will replace parameters)", async () => {
```

Comment:
This looks like the same test as https://github.com/pinterest/typespec/pull/50/files#r2436807470 except not `async`.

We don't need to test each scenario with both `async` and non-`async` versions. Testing that `async` works on its own is good enough.